tests/test_gist.py

```diff
@@ -236,6 +251,10 @@ def setUp(self):
         self.config.set('gist', 'gnupg-homedir', self.gnupghome)
         self.config.set('gist', 'gnupg-fingerprint', self.fingerprint)
 
+    @classmethod
+    def tearDownClass(cls):
```

Comment:
This test only uses a GPG agent if one is available -- it doesn't spawn or require one. So there is no guarantee that there is an agent running.

Reply:
I'm pretty sure that as of 2.1, gpg always uses the agent.  For instance, from the 2.1.15 man page,
> --no-use-agent
>              This is dummy option. gpg2 always requires the agent.

See also this issue from python-gnupg, https://github.com/isislovecruft/python-gnupg/issues/96

Reply:
I am going based upon the [documention|(https://pythonhosted.org/python-gnupg/) for the python-gnupg package. I think the repo you are looking at is an extensive fork of the original package. One of the main differences seems to be that the original is not using gpg2 but v1.4.9.

I'm still not sure why you need to make this change. Using your change that introduces `test_requires` I find that the tests are passing on a new system.

Reply:
Ah, bad link but the example is valid.  I'm assuming you're still using gnupg 1.x or something as python-gnupg rightly doesn't require specific version of gnupg.  The problem being fixed is that the unittests are leaving a gpg-agent laying around.  This has nothing to do with python-gnupg, the behavior of gpg itself changed in version 2.

Without the fix:
```
$ python setup.py test
running test
running egg_info
...
$ ps x | grep gpg-agent
32308 ?        Ss     0:00 gpg-agent --homedir /home/jbronder/scm/gist/tests/gnupg --use-standard-socket --daemon
```

Reply:
Ah I see. Unfortunate. On my laptop I am running ubuntu 14.10 and I have been testing on an instance running ubuntu 16.04. On both of these, I believe python-gnupg is using gpg1. And I believe the default is ubuntu 14.04 on travis. In these cases, I believe that there is no gpg-agent left running after the tests, so the command to kill the agent is failing.

Reply:
For Ubuntu, gpg2 is available in the earlier releases as the `gpg2` package.  Looks like the base `gpg` package moves to 2.1.15 in 17.04 if I'm reading this correctly, https://packages.ubuntu.com/search?keywords=gnupg

Reply:
I do not want to force a non-python package install for users and I am not going to break existing installations of gist by forcing users to install gpg2. However, I don't think that is necessary to fix this issue. The problem as I see it is that the agent is being launched unintentionally by the test and your fix is trying to manage that when it is not cleaned up. An alternative, which I would prefer, is that the agent is started intentionally (if possible) and stopped using a context manager. One thing is that the gpg-agent binary may not exist, so that situation also needs to be accommodated. I don't mind implementing that fix if you do not want to.

Reply:
Up to you.  I figured that it might just be sufficient to kill the started agent if it's there and otherwise just ignore the error if it's not there.  By the time distros are forcing gpg-2, they'll probably be past 2.1.13 and the `homedir` argument to `gpgconf` will work.

Anyways, I fixed my last commit so that tests are passing both with gpg2 and in your Travis setup.

Reply:
The problem that I am seeing when I run this is,

```
======================================================================
ERROR: tearDownClass (test_gist.TestGistGPG)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/gist/tests/test_gist.py", line 256, in tearDownClass
    kill_gpg_agent(cls.gnupghome)
  File "/root/gist/tests/test_gist.py", line 45, in kill_gpg_agent
    subprocess.call(args)
  File "/usr/lib/python2.7/subprocess.py", line 523, in call
    return Popen(*popenargs, **kwargs).wait()
  File "/usr/lib/python2.7/subprocess.py", line 711, in __init__
    errread, errwrite)
  File "/usr/lib/python2.7/subprocess.py", line 1343, in _execute_child
    raise child_exception
OSError: [Errno 2] No such file or directory

----------------------------------------------------------------------
Ran 8 tests in 0.151s

FAILED (errors=1)
Test failed: <unittest.runner.TextTestResult run=8 errors=1 failures=0>
error: Test failed: <unittest.runner.TextTestResult run=8 errors=1 failures=0>
```

This is on an instance of ubuntu 16.04 and there is no gpg-agent binary.